### PR TITLE
streamline auto bind

### PR DESCRIPTION
**Pull request by mafintosh**
Streamlines the bind, so A) the socket ALWAYS eagerly binds in ready, B) you can await this process and C) makes it easier to specify what port you prefer to bind to in general

### CODE REVIEW

File: lib/rpc.js [hunk 1]
```diff
@@ -45,31 +48,41 @@ module.exports = class RPC {
...
+      s.bind(port)
       s.on('listening', onlistening)
       s.on('error', onerror)
 
       function onlistening () {
+        this._bound = true
+
         s.removeListener('listening', onlistening)
         s.removeListener('error', onerror)
         resolve()
```
Comment (davidmarkclements) [inline lib/rpc.js:68]: if we resolve the promise with the port, then `bind()` will return a promise that resolves to port as a sort confirmation, we could also use this internally to eventuall set _binding to the actual port (and then for debugging you can check that _bind and _binding match (unless _bind is 0)  (eg `return (this,_binding = await this._binding)`)

**Review by davidmarkclements — APPROVED** [review]
small "nice-to-have" suggestion, sgtm either way though

File: lib/rpc.js (hunk 1 above)
Comment (mafintosh) [reply lib/rpc.js:68]: great idea, will make it resolve to the port

**Review by davidmarkclements — APPROVED** [review]
awesome

**Pull request merged**